GlotPress/GlotPress-WP
# Local inline translation

## What?
This adds inline translation capability to WordPress making use of the local GlotPress install.

## Why?
It speeds up translation :) It allows verifying translations in context.

## How?
This uses an adapted version of Automattic's [Community Translator](https://github.com/Automattic/community-translator). An important change is that the translations are fetched inside the PHP call and not through the `query-by-originals` call. This is possible since GlotPress lives on the same WordPress install.

## Testing Instructions
Saving a translation doesn't work yet. It depends on the [gp-translation-extended-api Plugin](https://github.com/akirk/gp-translation-extended-api).

Activate the translation tool using the globe. Observe translatable strings lighting up and right click them to translate them.

## Screenshots or screencast 
Activate
![Screenshot 2023-02-22 at 15 08 23](https://user-images.githubusercontent.com/203408/220646163-551e368e-8cac-4a0f-b82b-e40777b0ac5d.png)
Deactivate
![Screenshot 2023-02-22 at 15 08 26](https://user-images.githubusercontent.com/203408/220646196-77e3fdb6-e2a9-4d92-84e7-8522a70f921c.png)

![Screenshot 2023-02-21 at 15 37 57](https://user-images.githubusercontent.com/203408/220374753-581e6875-c4d6-4b40-8d38-bc70af49056e.png)

![Screenshot 2023-02-21 at 15 38 33](https://user-images.githubusercontent.com/203408/220374890-fc2e5c96-2485-4890-abeb-10c36008ec28.png)

When saving is accepted by the [gp-translation-extended-api Plugin](https://github.com/akirk/gp-translation-extended-api), that translation is maintained after reloads.
![local translations](https://user-images.githubusercontent.com/203408/220644512-b7d8ef58-8f1b-4897-ab22-987d291e23c8.gif)

Untranslatable strings (because the GlotPress project does not exist) expose their text domain on hover.
![Screenshot 2023-02-22 at 15 06 38](https://user-images.githubusercontent.com/203408/220644565-54bf5477-4b39-46b9-a442-bc584442d5d1.png)
